include/envoy/event/dispatcher.h

```diff
@@ -110,6 +110,14 @@ class Dispatcher {
                                               Network::ListenerCallbacks& cb, bool bind_to_port,
                                               bool hand_off_restored_destination_connections) PURE;
 
+  /**
+   * Create a logical udp listener on a specific port.
+   * @param socket supplies the socket to listen on.
+   * @param cb supplies the udp listener callbacks to invoke for listener events.
+   * @return Network::ListenerPtr a new listener that is owned by the caller.
+   */
+  virtual Network::ListenerPtr createDatagramListener(Network::Socket& socket,
```

Comment:
I was about to ask that we be consistent with Udp vs Datagram when I realized you were totally doing this right per mpw@'s advice.
I think when we get to docs it's worth doccing up that this is meant to be generic (potentially extending to datagrams over sockets) where the listener is for UdpSockets for others who have the same question :-P